### PR TITLE
refactor: Ensure support for `warn` mode for modern deprecations

### PR DESCRIPTION
**Pull request by pgrzesik**
Reported internally

### CODE REVIEW

**Review by medikoo — COMMENTED** [review]
Looks good, still, if I look correctly it doesn't cover well the scenario where just 1 deprecation was reported in summary mode. As then regular `writeDeprecation` function is used, and we've just now introduced modern logs to it.

So in modern logs on, with 1 deprecation, user will see both, a `sls doctor` notice and deprecation warning.

I think the only way to mitigate it is to rely on `SLS_DEV_LOG_MODE` and issue that `log.warning` only when have modern mode confirmed

**Comment by pgrzesik** [comment]
Good catch - I didn't test that scenario and didn't catch it

File: lib/utils/logDeprecation.js [hunk 1]
```diff
@@ -165,7 +172,7 @@ module.exports.printSummary = async () => {
     ]);
     await safeMoveFile(tmpHealthStatusFilename, healthStatusFilename);
 
-    if (bufferedDeprecations.length === 1) {
+    if (bufferedDeprecations.length === 1 && !isModernStyle) {
```
Comment (medikoo) [inline lib/utils/logDeprecation.js:175]: More accurate would be to add it at L80. Theoretically both modern and legacy logs could be outputted, and in this form, it'll prevent the writing of legacy logs in such case

Comment (pgrzesik) [reply lib/utils/logDeprecation.js]: Good point - I've changed the approach - if you see a better solution please let me know

File: lib/utils/logDeprecation.js [hunk 2]
```diff
@@ -55,22 +55,29 @@ const resolveMode = (serviceConfig) => {
...
+      `More Info: https://www.serverless.com/framework/docs/deprecations/#${code}`
+    );
     legacy.write(
       `            ${chalk.keyword('orange')(
         `More Info: https://www.serverless.com/framework/docs/deprecations/#${code}`
       )}\n`
     );
   }
+  if (!onlyLegacy) {
+    log.warning(modernLogsMessageLines.join('\n'));
```
Comment (medikoo) [inline lib/utils/logDeprecation.js:79]: It'll be nicer to go with what you proposed earlier, but just add this condition here as: `if (isModern) log.warning`

It'll also make update minimal in v3. 

(Passing it through option in internal function, where answer is visible upfront looks a bit overkill)

Comment (medikoo) [reply lib/utils/logDeprecation.js:79]: Indeed, you're right. Let's keep it the way you've proposed, sorry for confusion

**Review by medikoo — APPROVED** [review]
Looks great 👍

**Pull request merged**